src/rgw/rgw_rest_s3.cc

```diff
@@ -2395,8 +2395,8 @@ static void dump_bucket_metadata(req_state *s, rgw::sal::Bucket* bucket,
   dump_header(s, "X-RGW-Object-Count", static_cast<long long>(stats.num_objects));
   dump_header(s, "X-RGW-Bytes-Used", static_cast<long long>(stats.size));
 
-  // only bucket's owner is allowed to get the quota settings of the account
-  if (s->auth.identity->is_owner_of(bucket->get_owner())) {
+  // only bucket's owner is allowed to get the quota settings of the account if quota is enabled
+  if (s->auth.identity->is_owner_of(bucket->get_owner()) && s->bucket->get_info().quota.enabled) {
```

Comment:
`s->bucket->get_info().quota.enabled` should only control whether we dump the Bucket-Size/Bucket-Objects headers

there's a separate `user_info.quota.user_quota.enabled` for the User-Size/User-Objects headers

the Max-Buckets header isn't controlled by either of those, so should always be dumped

Reply:
@cbodley added enabled check for user and bucket info before dump headers : Bucket-Size/Bucket_Objects and User-Size/User-Objects.

Please review again.